docs/usage/7-middleware/3-builtin-middlewares/5-session-middleware.md

```diff
@@ -1,21 +1,188 @@
-    install it.
-
-To use the `SessionMiddleware` simply create an instance of
-[SessionCookieConfig][starlite.middleware.session.SessionCookieConfig] and pass the created middleware to any layer of
-the application:
-
-```py title="Hello World"
---8<-- "examples/middleware/session_middleware.py"
+    This requires the `aiomamcache` package. To install it you can install starlite with
+    `pip install starlite[memcached]`
+
+```py title="memcached_backend.py"
+--8<-- "examples/middleware/session/memcached_backend.py"
 ```
 
-For additional configuration options please see the [configuration references][starlite.middleware.session.SessionCookieConfig].
+!!! tip "See also"
+    - [Accessing the storage backend directly](#accessing-the-storage-backend-directly)
+    - [BaseBackendConfig][starlite.middleware.session.base.BaseBackendConfig]
+    - [ServerSideSessionConfig][starlite.middleware.session.base.ServerSideSessionConfig]
+
+### In-memory storage
+
+The [Memory backend][starlite.middleware.session.memory_backend.MemoryBackend] can store
+session data in memory.
+
+!!! Danger
+    This should not be used in production. It primarily exists as a dummy backend for
+    testing purposes. It is not process safe, and data will not be persisted.
+
+```py title="memory_backend.py"
+--8<-- "examples/middleware/session/memory_backend.py"
+```
+
+### Database storage
+
+Database storage is currently offered through the
+[SQLAlchemyBackend][starlite.middleware.session.sqlalchemy_backend.BaseSQLAlchemyBackend].
+It supports both sync and async-engines and integrates with the [SQLAlchemyPlugin](/starlite/usage/10-plugins/1-sql-alchemy-plugin).
+Expired sessions will only be deleted when trying to access them. They can be manually deleted using the
+[delete_expired][starlite.middleware.session.sqlalchemy_backend.BaseSQLAlchemyBackend.delete_expired] method.
+
+There are two backends for SQLAlchemy:
+
+- [SQLAlchemyBackend][starlite.middleware.session.sqlalchemy_backend.SQLAlchemyBackend] for synchronous engines
+- [AsyncSQLAlchemyBackend][starlite.middleware.session.sqlalchemy_backend.AsyncSQLAlchemyBackend] for asynchronous engines
+
+When using the [configuration][starlite.middleware.session.sqlalchemy_backend.SQLAlchemyBackendConfig] object,
+it will automatically pick the correct backend to use based on the engine configuration.
+
+!!! info
+    This requires [sqlalchemy](https://sqlalchemy.org/). You can install it via
+    `pip install sqlalchemy`.
+
+=== "Synchronous engine"
+
+    ```py title="sqlalchemy_backend.py"
+    --8<-- "examples/middleware/session/sqlalchemy_backend.py"
+    ```
+
+=== "Asynchronous engine"
+
+    ```py title="sqlalchemy_backend.py"
+    --8<-- "examples/middleware/session/sqlalchemy_backend_async.py"
+    ```
+
+#### Supplying your own session model
+
+If you wish to extend the built-in session model, you can subclass the
```

Comment:
This might be a very dumb question but I have no significant experience with SQLAlchemy here so forgive me if the answer is obvious.

How is the user supposed to create session table in their DB for the built-in session model?

Reply:
Not a dumb question, but can you clarify a bit?

The built-in model is just a regular SQLAlchemy declarative mapping. 

These two examples are functionally equivalent:

```python
Base = declarative_base()
SessionModel = create_session_model(Base)
```

```python
Base = declarative_base()

class SessionModel(Base, SessionModelMixin):
    __tablename__ = "session"
    id = Column(Integer, primary_key=True)
```

Maybe it would help if this was explained in more detail in the docs?

Reply:
@provinzkraut usually the user needs to create a migration in their migration tool (I assume common choice for SQLAlchemy is alembic). So atm I don' understand if the user needs to do this for builtin session model or not.

If they don't need to do this because SQLAlchemy will do this then it would be great mention it in the docs. If they need to add a migration then this also worth mentioning and probably the docs should contain table structure.

Reply:
Ah, I see. Thanks for the clarification!

So, the SQLAlchemyBackend doesn't do anything in regards to migrations. It only provides a convenient way to create a model. The rest is up to the user to integrate it into their process. I'll mention this and update the example to be "run as-is".

Reply:
Added!